ui/apps/platform/src/utils/type.utils.ts

```diff
@@ -53,6 +53,10 @@ export function tupleTypeGuard<const T extends readonly string[]>(
 
 export type UnionFrom<T extends readonly string[]> = T[number];
 
-export type DeepPartial<T> = {
-    [P in keyof T]?: T[P] extends object ? DeepPartial<T[P]> : T[P];
+export type DeepPartialByKey<T, K extends string> = {
+    [P in keyof T]: P extends K
+        ? Partial<T[P]>
```

Comment:
Nit: do we want this to be deeply recursive, or only `Partial` the found keys at their first depth?

We could do infinite partial with this but it probably isn't necessary (and certainly isn't necessary for the current use case).
```diff
-         ? Partial<T[P]>
+         ? Partial<DeepPartialByKey<T[P], K>>
```

I bet we could even accept a third type parameter to specific the depth, but that seems more like a fun exercise than something we would actually use 😃

Reply:
Yeah, that's a good point! Adding another element to determine depth would be helpful. I think the current implementation is sufficient for now. We can revisit it, if necessary 👍🏼